lib/core/src/display/shape/text/text_field/word_occurrence.rs

```diff
@@ -0,0 +1,202 @@
+
+impl WordRange {
+    /// Creates a new `WordRange`.
+    pub fn new(word_range : Range<TextLocation>) -> Self {
+        let is_selected = false;
+        Self {word_range,is_selected}
+    }
+
+    /// Returns a word under cursor, if any.
+    pub fn word_at_cursor(content:&TextFieldContent, cursor:&Cursor) -> Option<Self> {
+        let line  = cursor.position.line;
+        let chars = content.lines()[line].chars();
+        if let Some(range) = get_index_range_of_word_at(chars,cursor.position.column) {
+            let column = range.start;
+            let start  = TextLocation {line,column};
+            let column = range.end;
+            let end    = TextLocation {line,column};
+            Some(Self::new(start..end))
+        } else {
+            None
+        }
+    }
+}
+
+
+
+// =======================
+// === WordOccurrences ===
+// =======================
+
+/// A struct containing word occurrences in `TextFieldContent`.
+#[derive(Shrinkwrap,Clone,Debug)]
+pub struct WordOccurrences {
+    /// Words occurrences.
+    #[shrinkwrap(main_field)]
+    pub words     : Vec<WordRange>,
+    current_index : usize
+}
+
+impl WordOccurrences {
+    /// Gets the range of each occurrence of word.
+    pub fn new(content:&TextFieldContent, cursor:&mut Cursor) -> Option<Self> {
+        let range = if cursor.has_selection() {
+            Some(cursor.selection_range())
+        } else {
+            let word = WordRange::word_at_cursor(content, &cursor);
+            word.map(|word| Some(word.word_range)).unwrap_or(None)
+        };
+        if let Some(range) = range {
+            let word = content.copy_fragment(range);
+            let cursor_location = cursor.position;
+            let mut words = Vec::new();
+            let word: Vec<char> = word.chars().collect();
+
+            for (index, line) in content.lines().iter().enumerate() {
+                let line_chars = line.chars();
+                let words_in_line = get_word_occurrences(line_chars, &word);
+                for word in words_in_line {
+                    let line = index;
+                    let column = word.start;
+                    let start = TextLocation { line, column };
+                    let column = word.end;
+                    let end = TextLocation { line, column };
+                    words.push(WordRange::new(start..end));
+                }
+            }
+
+            if words.is_empty() {
+                None
+            } else {
+                let current_index      = words.iter().find_position(|current_word| {
+                    cursor_location.line == current_word.start.line &&
+                        cursor_location.column >= current_word.start.column &&
+                        cursor_location.column <= current_word.end.column
+                }).map(|(index, _)| index).unwrap_or(0);
+                let current_index = current_index - 1;
+                Some(Self { words, current_index }.initialize(&cursor))
+            }
```

Comment:
use a function for convertong bool to maybe from boolinator (Exported in prelude)

Reply:
I used filter instead :)